tsfresh/feature_selection/selection.py

```diff
@@ -124,7 +128,20 @@ def select_features(X, y, test_for_binary_target_binary_feature=defaults.TEST_FO
 
         y = pd.Series(y, index=X.index)
 
-    df_bh = check_fs_sig_bh(X, y, n_jobs, chunksize, fdr_level, hypotheses_independent,
-                            test_for_binary_target_real_feature)
-
-    return X.loc[:, df_bh[df_bh.rejected].Feature]
+    if multiclass is True:
+        unique_labels = y.unique()
+        if len(unique_labels) < 3:
+            raise RuntimeError("Cannot perform multiclass selection on binary target")
```

Comment:
0.9.0 will have some breaking changes anyway, so now is the time to change the API.

I like your proposal.

I would default that parameter to "None", meaning that the method tries to infer the type of target by looking at the dtype. I would not only look for type `int` but also look for targets that can be casted to int, something like `[1.0, 0.0, 1.0]`. 

But, I am not sure if `selection` is the right name. Maybe something like `ml_task`?